flutter_local_notifications/lib/src/callback_dispatcher.dart

```diff
@@ -28,7 +28,21 @@ void callbackDispatcher() {
         .map<Map<String, dynamic>>(
             (Map<dynamic, dynamic> event) => Map.castFrom(event))
         .listen((Map<String, dynamic> event) {
-      callback?.call(event['id'], event['input'], event['payload']);
+      final Object notificationId = event['notificationId'];
+      final int id;
+      if (notificationId is int) {
```

Comment:
Wondering if we can use a `try catch` or `int.tryParse` or similar? Also, `int.parse` may throw

Reply:
Not sure how that would help.
Maybe a better question is why are the notification ids integers in the first place, both iOS and android have String identifiers

Reply:
I was thinking:

```
try {
  id = int.parse(notificationId);
} catch(..) {
  id = notificationId as int;
}
```

However, perhaps this can not be written any shorter, lets leave it at that.

Reply:
That doesnt convey the intent though

Reply:
Yes you are right. Thx!